bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -282,6 +279,95 @@ static int callAndWait(String[] pstr, ToIntFunction<IUnknown> callable) {
 	return phr[0];
 }
 
+class WebViewProvider {
+
+	private CompletableFuture<ICoreWebView2> webViewFuture = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_2> webView_2Future = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_11> webView_11Future = new CompletableFuture<>();
+
+	private CompletableFuture<Void> lastWebViewTask = webViewFuture.thenCompose(__ -> {
+		return CompletableFuture.allOf(webView_2Future.handle((result, ex) -> null), webView_11Future.handle((result, ex) -> null));
+	});
+
+
+	private ICoreWebView2 getWebView(boolean waitForPendingWebviewTasksToFinish) {
+		if(waitForPendingWebviewTasksToFinish) {
+			waitForPendingWebviewTasksToFinish();
+		}
+		return webViewFuture.join();
+	}
+
+	private void initializeWebView(ICoreWebView2 webView) {
+		webViewFuture.complete(webView);
+	}
+
+	private boolean isWebViewUnavailable() {
+		return webViewFuture.isCancelled();
+	}
+
+	private ICoreWebView2_2 getWebView_2(boolean waitForPendingWebviewTasksToFinish) {
+		if(waitForPendingWebviewTasksToFinish) {
+			waitForPendingWebviewTasksToFinish();
+		}
+		return webView_2Future.join();
+	}
+
+	private void initializeWebView_2(ICoreWebView2_2 webView_2) {
+		webView_2Future.complete(webView_2);
+	}
```

Comment:
Like for the ICoreWebView itself, I propose to adapt the ones for WebView_2 and WebView_11 to also have the initialization logic within the provider. Then you can even get rid of the `cancelWebView...` etc. functions which are only part of the initialization and should be encapsulated in this provider class.